### PR TITLE
Added Quantization 8 bit support for Relu

### PR DESCRIPTION
**Pull request by amitsrivastava78**
This is asked by many in the community.

### CODE REVIEW

**Comment by amitsrivastava78** [comment]
@jianlijianli, I have rebased the code and resolved the merge conflicts, can you please review the PR.

Regards
Amit

File: tensorflow/lite/kernels/activations.cc [hunk 1]
```diff
@@ -328,6 +328,19 @@ TfLiteStatus PreluPrepare(TfLiteContext* context, TfLiteNode* node) {
   return kTfLiteOk;
 }
 
+template <typename T>
+void QuantizedRelu(const TfLiteTensor* input, TfLiteTensor* output) {
+  ActivationParams params;
+  params.activation_type = FusedActivationFunctionType::kRelu;
+  params.quantized_activation_min =
```
Comment (jianlijianli) [inline tensorflow/lite/kernels/activations.cc]: Could you please explain why is zero point used in min but not in max?

**Review by jianlijianli — COMMENTED** [review]
Could you please update register.cc, operator.cc and operator_test.cc as well, same as in https://github.com/tensorflow/tensorflow/commit/8d4cdf8444389286ec54b7105b07e9059dba4f66 ? Thanks.

**Comment by jianlijianli** [comment]
Thanks for working on this and apologies for the delay. Please see my comment.

File: tensorflow/lite/kernels/activations.cc (hunk 1 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/activations.cc]: @jianlijianli , thanks for the review, i have updated the code, kindly check.

Regards
Amit

**Comment by amitsrivastava78** [comment]
> Could you please update register.cc, operator.cc and operator_test.cc as well, same as in [8d4cdf8](https://github.com/tensorflow/tensorflow/commit/8d4cdf8444389286ec54b7105b07e9059dba4f66) ? Thanks.

@jianlijianli , have updated the code, as per your suggestion, kindly check.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli , the implementation is also validated against #28268 (issue) . Can you please check.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli , i have added one more commit for the support of Relu from tools as well.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli , could you please have a look and provide your feedback for the same.This is one of the important PRs for me and would like to see this though.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli i have updated the registery for the Relu kernel with updated version, kindly review.

Regards
Amit

**Review by jianlijianli — COMMENTED** [review]
Hi Amit, thanks for making the changes. Apologies if it was misleading, by "operator.cc" I mean "tensorflow/lite/toco/tflite/operator.cc" not "tensorflow/lite/tools/optimize/operator_property.cc". Could you please add the versioning test in "tensorflow/lite/toco/tflite/operator.cc", like in here (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/toco/tflite/operator.cc#L278-L286)? Thanks.

Also, thanks for working on operator_property.cc. I added a comment that for ReLU family there is a restriction on input and output scale. This change might require you to update test on "relu.bin". Please feel free to leave all the changes related to "tensorflow/lite/tools/optimize/operator_property.cc" to a different PR. Thanks!

File: tensorflow/lite/tools/optimize/operator_property.cc [hunk 2]
```diff
@@ -25,6 +25,11 @@ OperatorProperty GetOperatorProperty(const BuiltinOperator& op) {
       property.output_indexes = {0};
       property.version = 2;
       break;
+    case BuiltinOperator_RELU:
```
[the remark this answers is not in the document]
Comment (amitsrivastava78) [reply tensorflow/lite/tools/optimize/operator_property.cc]: @jianlijianli , thanks for the comments, i have updated as per your suggestion, kindly check.

Regards
Amit

**Comment by amitsrivastava78** [comment]
> Hi Amit, thanks for making the changes. Apologies if it was misleading, by "operator.cc" I mean "tensorflow/lite/toco/tflite/operator.cc" not "tensorflow/lite/tools/optimize/operator_property.cc". Could you please add the versioning test in "tensorflow/lite/toco/tflite/operator.cc", like in here (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/toco/tflite/operator.cc#L278-L286)? Thanks.
> 
> Also, thanks for working on operator_property.cc. I added a comment that for ReLU family there is a restriction on input and output scale. This change might require you to update test on "relu.bin". Please feel free to leave all the changes related to "tensorflow/lite/tools/optimize/operator_property.cc" to a different PR. Thanks!

@jianlijianli , Thanks for spending time on the PR i have updated the code as per your suggestions, also i have added the TC as well. Kindly check and approve.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli can you please check and approve the PR.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli , i have resolved the merge conflict , can you please check and approve.


Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli , sorry to bug you again, i have rebased the code again, would be great if you can spend some time on this PR.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@jianlijianli , this is a long pending PR, can you please take a decision on this PR.

Regards
Amit

**Comment by rthadur** [comment]
@amitsrivastava78 thanks for your patience , it will be reviewed soon.

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by gbaned** [comment]
@amitsrivastava78  Can you please resolve the conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 44 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Comment by InCogNiTo124** [comment]
What's the status of this?

**Comment by karimnosseir** [comment]
@InCogNiTo124  The support for Quantized relu was added already.

**Pull request closed without merging**